### ISSUE TITLE
bevy_utils::default abbreviation lacks warning

### ISSUE THREAD

**Issue by Leonss23** [bevyengine/bevy#8879]
## Bevy version

0.10.1

## What you did

```rust
struct SomeStruct;
impl Default for SomeStruct {
    fn default() -> Self {
        Self {
            // ...fields,
            ..default()
        }
    }
}
```

## What went wrong

The previous code overflows the stack without warning.

## Additional information

If `..Default::default()` is used instead of `..default()`, the function gives a warning `function cannot return without recursing`.

**Comment by alice-i-cecile** [comment]
Hmm, I wonder if this is something we can fix ourselves. I suspect that this is ultimately blocked on changes upstream in `rustc` to how these warnings are generated.

**Comment by alice-i-cecile** [comment]
Still very useful to have this issue: it'll show up in the search results!

**Comment by mockersf** [comment]
https://github.com/rust-lang/rust/issues/73014#issuecomment-1597322434

**Comment by harudagondi** [comment]
This is known as a Rust issue.

- rust-lang/rust#57965

**Comment by RylanYancey** [comment]
This issue is still occuring in 1.15.2

**Comment by olekspickle** [comment]
sheesh, just encountered it. nasty bugger

**Comment by may-bell** [comment]
still occurring:
- bevy 0.18.1
- arch linux
- kate with rust-analyzer

very grateful for this issue page instructing me to use `..Default::default()` so the rust-analyzer can find the recursion.

this is dreadfully easy to overflow the stack for a beginner (me :D), especially since the docs suggested i use `..default()` instead of `..Default::default()` when implementing the Default trait for my components. a warning would be really nice, but it sounds like this is dependent on the rust team?

**Comment by alice-i-cecile** [comment]
Yep, not within our control, unless we want to remove the `default()` abbreviation completely. Which, TBH, we could probably do now that bsn! is shipping for a better approach to implicit defaults.
